demo/run-osm-demo.sh

```diff
@@ -86,6 +87,9 @@ fi
 
 if [ "$PUBLISH_IMAGES" = true ]; then
     make docker-push
+fi 
+
+if [ "$USE_PRIVATE_REGISTRY" = true ]; then
     ./scripts/create-container-registry-creds.sh "$K8S_NAMESPACE"
```

Comment:
You need to also do the same thing that you do here in the configure-app-namespaces script.

Reply:
I address that in this PR: https://github.com/openservicemesh/osm/pull/3876

Reply:
^ realized I need to update the default value of USE_PRIVATE_REGISTRY there to false as well